test/bigtable/table.js

```diff
@@ -0,0 +1,852 @@
+          assert.strictEqual(formatSpy.callCount, 1);
+          assert.strictEqual(formatSpy.getCall(0).args[0], options);
+          done();
+        };
+
+        table.getRows(options);
+      });
+
+      it('should retrieve multiple rows', function(done) {
+        var options = {
+          keys: [
+            'gwashington',
+            'alincoln'
+          ]
+        };
+
+        table.requestStream = function(g, reqOpts) {
+          var convertSpy = FakeMutation.convertToBytes;
+
+          assert.deepEqual(reqOpts.rowSet.rowKeys, options.keys);
+          assert.strictEqual(convertSpy.callCount, 2);
+          assert.strictEqual(convertSpy.getCall(0).args[0], options.keys[0]);
+          assert.strictEqual(convertSpy.getCall(1).args[0], options.keys[1]);
+          done();
+        };
+
+        table.getRows(options);
+      });
+
+      it('should retrieve multiple ranges', function(done) {
+        var options = {
+          ranges: [{
+            start: 'a',
+            end: 'b'
+          }, {
+            start: 'c',
+            end: 'd'
+          }]
+        };
+
+        table.requestStream = function(g, reqOpts) {
+          assert.deepEqual(reqOpts.rowSet.rowRanges, options.ranges);
+          assert.strictEqual(formatSpy.callCount, 2);
+          assert.strictEqual(formatSpy.getCall(0).args[0], options.ranges[0]);
+          assert.strictEqual(formatSpy.getCall(1).args[0], options.ranges[1]);
+          done();
+        };
+
+        table.getRows(options);
+      });
+
+      it('should parse a filter object', function(done) {
+        var options = {
+          filter: [{}]
+        };
+
+        table.requestStream = function(g, reqOpts) {
+          var parseSpy = FakeFilter.parse;
+
+          assert.strictEqual(reqOpts.filter, options.filter);
+          assert.strictEqual(parseSpy.callCount, 1);
+          assert.strictEqual(parseSpy.getCall(0).args[0], options.filter);
+          done();
+        };
+
+        table.getRows(options);
+      });
+
+      it('should allow row interleaving', function(done) {
+        var options = {
+          interleave: true
+        };
+
+        table.requestStream = function(g, reqOpts) {
+          assert.strictEqual(reqOpts.allowRowInterleaving, options.interleave);
+          done();
+        };
+
+        table.getRows(options);
+      });
+
+      it('should allow setting a row limit', function(done) {
+        var options = {
+          limit: 10
+        };
+
+        table.requestStream = function(g, reqOpts) {
+          assert.strictEqual(reqOpts.numRowsLimit, options.limit);
+          done();
+        };
+
+        table.getRows(options);
+      });
+    });
+
+    describe('success', function() {
+      var fakeRows = [{
+        rowKey: 'a',
+        chunks: []
+      }, {
+        rowKey: 'b',
+        chunks: []
+      }];
+
+      beforeEach(function() {
+        sinon.stub(table, 'row', function(name) {
+          return new Row(name);
```

Comment:
We should just return some random object that we confirm is the same one received later.